cluster/examples/kubernetes/rook-operator.yaml

```diff
@@ -122,3 +128,17 @@ spec:
         # current mon with a new mon (useful for compensating flapping network).
         - name: ROOK_MON_OUT_TIMEOUT
           value: "300s"
+        - name: ROOK_AGENT_IMAGE_TAG
+          value: master
+        - name: ROOK_SYSTEM_NAMESPACE
+          valueFrom:
+            fieldRef:
+              fieldPath: metadata.namespace
+        - name: NODE_NAME
+          valueFrom:
+            fieldRef:
+              fieldPath: spec.nodeName
+        - name: ROOK_OPERATOR_SERVICE_ACCOUNT
+          valueFrom:
+            fieldRef:
+              fieldPath: spec.serviceAccountName
```

Comment:
Node_name is needed on the operator because it needs to talk to the node api to determine where the flexvolume driver should be placed. So that the agents can host mount this directory when starting up. Here is the code. https://github.com/kokhang/rook/blob/rook-flex/pkg/operator/agent/agent.go#L121

Reply:
About the service account, I had a different perspective. I see Rook as a single software/application and i would assume one service account for the `rook` software. @jbw976 and I had a discussion about this. I believe you and him are advocating for separate service account for each components (api, osd, agents, operator, etc). Im ok if you guys prefer this way.